framework/src/base/ComputeUserObjectsThread.C

```diff
@@ -73,23 +73,21 @@ void
 ComputeUserObjectsThread::onElement(const Elem * elem)
 {
   _fe_problem.prepare(elem, _tid);
```

Comment:
Before this is merged we'll document this in the source so it's not a point of confusion. The reason this is here is because this loop is activated for elements, side, and nodal `UserObjects`. The volumetric prepare is necessary for all three even if we don't have any elemental UOs. The other two methods for preparing materials however are not and @YaqiWang already moved those. 

I'd like to see a test made for block restricted UOs before this is merged. I did timings on several existing UO tests (including those with just side UOs) and saw no speedup at all even after scaling the problem to run for over a minute on my mac. I'm sure that this is indeed an optimization but I'd feel better if we could at least prove it.